File: app/src/main/java/com/example/keybindhelper/Dialogs/ConfirmDialog.java
package com.example.keybindhelper.Dialogs;

import android.app.Dialog;
import android.content.Context;
import android.widget.TextView;

import com.example.keybindhelper.R;

public class ConfirmDialog {
    public ConfirmedEvent onConfirmed;
    private Dialog dialog;
    public ConfirmDialog(Context c,String title){
        dialog=new Dialog(c);
        dialog.setContentView(R.layout.confirm_dialog);
        ((TextView)dialog.findViewById(R.id.confirm_dialog_title)).setText(title);
        dialog.findViewById(R.id.confirm_dialog_cancel_btn).setOnClickListener(v->{
            dialog.cancel();
        });
        dialog.findViewById(R.id.confirm_dialog_ok_btn).setOnClickListener(v->{
            dialog.cancel();
            onConfirmed.onConfirmed();
        });
    }
    public void Show(){
        dialog.show();
    }
    public interface ConfirmedEvent{
        void onConfirmed();
    }
}

File: app/src/main/java/com/example/keybindhelper/Dialogs/ConfirmDialog.kt
package com.example.keybindhelper.Dialogs

import android.app.Dialog
import android.content.Context
import android.view.View
import com.example.keybindhelper.R
import android.widget.TextView

class ConfirmDialog(c: Context?, title: String?) {
    var onConfirmed: ConfirmedEvent? = null
    private val dialog: Dialog
    fun Show() {
        dialog.show()
    }

    interface ConfirmedEvent {
        fun onConfirmed()
    }

    init {
        dialog = Dialog(c!!)
        dialog.setContentView(R.layout.confirm_dialog)
        (dialog.findViewById<View>(R.id.confirm_dialog_title) as TextView).text = title
        dialog.findViewById<View>(R.id.confirm_dialog_cancel_btn)
            .setOnClickListener { v: View? -> dialog.cancel() }
        dialog.findViewById<View>(R.id.confirm_dialog_ok_btn).setOnClickListener {
            dialog.cancel()
            onConfirmed!!.onConfirmed()
        }
    }
}